src/main/java/org/takes/facets/auth/PsBasic.java

```diff
@@ -185,4 +190,159 @@ public Fake(final boolean cond) {
             return new Opt.Empty<Identity>();
         }
     }
+
+    /**
+     * Default implementation. Using list of user and pass.
+     * @author lcozzani (lautaromail@gmail.com)
+     * @version $Id$
+     * @since 0.22
+     */
+    public static final class Default implements PsBasic.Entry {
+
+        /**
+         * List of users and its passwords.
+         */
+        private final transient ConcurrentMap<String, PsBasic.User> users;
+
+        /**
+         * Ctor with list of "user,pass,urn".
+         * @param list List of users.
+         */
+        public Default(final String ... list) {
+            this(buildList(list));
+        }
+
+        /**
+         * Ctor.
+         * @param usrs Map of user and pass
+         */
+        public Default(final List<PsBasic.User> usrs) {
+            this.users = Default.buildMap(usrs);
+        }
+
+        @Override
+        public Opt<Identity> enter(final String user, final String pwd) {
+            final Opt<Identity> valid;
+            if (this.users.containsKey(user)
+                && this.users.get(user).pass().equals(pwd)) {
+                valid = new Opt.Single<Identity>(
+                    new Identity.Simple(this.users.get(user).urn())
+                );
+            } else {
+                valid = new Opt.Empty<Identity>();
+            }
+            return valid;
+        }
+
+        /**
+         * Build list of users.
+         * @param list List of users format "user,pass,urn"
+         * @return List {@link List} of users
+         */
+        @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+        private static List<PsBasic.User> buildList(final String[] list) {
+            final List<PsBasic.User> users =
+                new ArrayList<PsBasic.User>(list.length);
+            for (final String user : list) {
+                final PsBasic.User usr = new PsBasic.User(user);
+                users.add(usr);
+            }
+            return users;
+        }
+
+        /**
+         * Build Map of users.
+         * @param usrs List {@link List} of users.
+         * @return Map {@link Map} of users.
+         */
+        private static ConcurrentMap<String, PsBasic.User> buildMap(
```

Comment:
@lautarobock same here, see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#method-names